File: Army/rogue.cpp
#include "rogue.h"

using namespace std;

Rogue::Rogue(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage) {}

Rogue::~Rogue() {}

void Rogue::attack(Unit* victim) {
    ensureIsAlive();
    victim->takeDamage(getDamage());
}

File: Java/src/com/gymfox/army/units/Rogue.java
package com.gymfox.army.units;

import com.gymfox.army.ability.RogueAbility;
public class Rogue extends Unit {
    public Rogue(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new RogueAbility(this);
    }
}
